.gitlab-ci.yml

```diff
@@ -96,38 +98,63 @@ sniffers:sonarqube:
   - master
   allow_failure: true
 
-prepare:front:
-  cache:
-    key: ${CI_COMMIT_REF_SLUG}-front
-    paths:
-      - ${THEME_PATH}/node_modules/
+
+prepare:back:
   stage: prepare
   script:
-  - make front
+    - docker --version
+    - docker-compose --version
+    - echo "INSTALL_DEV_DEPENDENCIES=TRUE" >> .env.default
+    - make back
+    - docker-compose down # PHP container was required for composer
+  cache:
+    key:
+      files:
+        - composer.json
+        - composer.lock
+    paths:
+      - vendor/
+      - web/core/
+      - web/libraries/
+      - web/modules/contrib/
+      - web/profiles/contrib/
+      - web/themes/contrib/
+      - drush/contrib/
+  dependencies: []
   artifacts:
-    name: "$CI_COMMIT_REF_NAME:$CI_COMMIT_SHA"
+    name: "$CI_COMMIT_REF_NAME:$CI_COMMIT_SHA:back"
     paths:
-      - ${THEME_PATH}/dist/
-  only:
-    variables:
-      - $THEME_PATH
+      - vendor/
+      - web/
+      - drush/contrib/
   <<: *ra_tags_branches
 
-.job_deploy_commit_template: &job_deploy_commit
-  stage: deploy
-  before_script:
-    - date; pwd; ls -lahq;
-    - echo "Clean-up build dir ${BUILD_DIR}"
-    # Always down previous environment.
-    - (if [ -d ${BUILD_DIR} ]; then date; cd ${BUILD_DIR}; pwd; make clean; cd -; rm -rf ${BUILD_DIR}; fi)
+prepare:front:
+  stage: prepare
   script:
-    - echo "Removed previous review app ${CI_ENVIRONMENT_URL} from ${BUILD_DIR}."
+    - make front-build
+  only:
+    variables:
+      - $THEME_PATH
+  cache:
+    key:
+      files:
+        - ${THEME_PATH}/package.json
+        - ${THEME_PATH}/yarn.lock
+    paths:
+      - ${THEME_PATH}/node_modules/ # Populated during yarn install
   dependencies:
-    - sniffers:phpcs
+    - sniffers:front
+  artifacts:
+    name: "$CI_COMMIT_REF_NAME:$CI_COMMIT_SHA:front"
+    paths:
+      - ${THEME_PATH}/node_modules/ # Populated during yarn install
```

Comment:
Do we really need to add node packages to artifacts? After this point we don't seem to need them anymore cause lint and build already passed.

Reply:
sliderInit

Reply:
Makes total sense ! After assets get compiled by `yarn build` > `node_modules` dir is not useful for RA, only `dist` dir is.
Gonna make a follow up PR